File: frontend/src/app/services/show.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Show, Seat } from '../shared/models/models';

import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ShowService {
  private apiUrl = `${environment.apiUrl}/shows`;

  constructor(private http: HttpClient) { }

  getShowsByMovie(movieId: number): Observable<Show[]> {
    return this.http.get<Show[]>(`${this.apiUrl}/movie/${movieId}`);
  }

  getShowById(id: number): Observable<Show> {
    return this.http.get<Show>(`${this.apiUrl}/${id}`);
  }

  getSeatsForShow(showId: number): Observable<Seat[]> {
    return this.http.get<Seat[]>(`${this.apiUrl}/${showId}/seats`);
  }
}

File: backend/src/main/java/com/icinema/service/ShowService.java
package com.icinema.service;

import com.icinema.dto.SeatDTO;
import com.icinema.dto.ShowDTO;
import java.util.List;

public interface ShowService {
    List<ShowDTO> getShowsByMovie(Long movieId);

    ShowDTO getShowById(Long id);

    List<SeatDTO> getSeatsForShow(Long showId);
}
